cpp/periodic_module/periodic.cpp

```diff
@@ -193,36 +246,189 @@ void ValidateBatchSize(const mgp::Value &batch_size_value) {
   }
 
-  const auto *maybe_port = std::getenv(kMgPort);
-  if (maybe_port) {
-    port = static_cast<uint16_t>(std::move(*maybe_port));
+  if (value.IsString()) {
+    return;
   }
 
-  const auto *maybe_username = std::getenv(kMgUsername);
-  if (maybe_username) {
-    username = std::move(maybe_username);
+  if (value.IsList()) {
+    auto list_value = value.ValueList();
+    for (auto elem : list_value) {
+      if (!elem.IsString()) {
+        throw std::runtime_error(fmt::format("Invalid config for config parameter {}!", config_key));
+      }
+    }
   }
+}
 
-  const auto *maybe_password = std::getenv(kMgPassword);
-  if (maybe_password) {
-    password = std::move(maybe_password);
+void ValidateDeletionConfig(const mgp::Map &config) {
+  auto batch_size_key = std::string(kConfigKeyBatchSize);
+  auto labels_key = std::string(kConfigKeyLabels);
+  auto edge_types_key = std::string(kConfigKeyEdgeTypes);
+
+  if (!config.KeyExists(batch_size_key)) {
+    throw std::runtime_error("Periodic.delete() did not specify config parameter batch_size!");
   }
 
-  return mg::Client::Params{.host = std::move(host),
-                            .port = std::move(port),
-                            .username = std::move(username),
-                            .password = std::move(password)};
+  auto batch_size_value = config.At(batch_size_key);
+  if (!batch_size_value.IsInt()) {
+    throw std::runtime_error("Batch size needs to be an integer!");
+  }
+
+  if (batch_size_value.ValueInt() <= 0) {
+    throw std::runtime_error("Batch size can't be a non-negative integer!");
+  }
+
+  ValidateDeletionConfigEntities(config, labels_key);
+  ValidateDeletionConfigEntities(config, edge_types_key);
+}
+
+void EmplaceFromConfig(const mgp::Map &config, std::vector<std::string> &vec, std::string &config_key) {
+  auto key = std::string_view(config_key);
+  if (config.KeyExists(key)) {
+    auto value = config.At(key);
+    if (value.IsString()) {
+      vec.emplace_back(std::string(value.ValueString()));
+    } else if (value.IsList()) {
+      auto list_value = value.ValueList();
+      for (const auto elem : list_value) {
+        vec.emplace_back(elem.ValueString());
+      }
+    }
+  }
+}
+
+DeletionInfo GetDeletionInfo(const mgp::Map &config) {
+  std::vector<std::string> labels, edge_types;
+
+  ValidateDeletionConfig(config);
+
+  auto batch_size_key = std::string(kConfigKeyBatchSize);
+  auto labels_key = std::string(kConfigKeyLabels);
+  auto edge_types_key = std::string(kConfigKeyEdgeTypes);
+
+  auto batch_size = config.At(batch_size_key).ValueInt();
+
+  EmplaceFromConfig(config, labels, labels_key);
+  EmplaceFromConfig(config, edge_types, edge_types_key);
+
+  return {.batch_size = static_cast<uint64_t>(batch_size),
+          .labels = std::move(labels),
+          .edge_types = std::move(edge_types)};
+}
+
+void ExecutePeriodicDelete(DeletionInfo deletion_info, DeletionResult &deletion_result) {
```

Comment:
You can accept here by const &, and then you don't need to move, as this way we need move ctor, and other with const &, it is just variable creation